common/data_refinery_common/models/sample.py

```diff
@@ -113,6 +113,14 @@ def to_metadata_dict(self):
         metadata["refinebio_annotations"] = [
             data for data in self.sampleannotation_set.all().values_list("data", flat=True)
         ]
+        metadata["refinebio_processor_info"] = [
+            {
+                "id": result.processor.id,
+                "name": result.processor.name,
+                "version": result.processor.version,
+            }
+            for result in self.results.filter(processor__isnull=False)
```

Comment:
I think some processors may create multiple results, so you should probably make sure to dedup the processors.

But that's actually probably not enough. This metadata ends up in a TSV file as well as a JSON file, so it needs to be flat for each sample. (The resulting TSV file is a matrix that has samples as rows and metadata values as columns or vice versa.) We really only want to include the processor information for the specific computed file we're using in the dataset/compendium: https://github.com/AlexsLemonade/refinebio/blob/dev/workers/data_refinery_workers/processors/smashing_utils.py#L77

However as you can see, which file is used is dependent on `job_context["dataset"].quant_sf_only`. Therefore this method will probably need to take in that boolean or the file itself, and then do something like: `metadata["refinebio_processor_id"] = computed_file.result.processor.id` for each of id, name, and version.

Reply:
Ok I've updated it to use the computed file to get the processor information